zap/src/main/java/org/zaproxy/zap/extension/search/SearchThread.java

```diff
@@ -279,6 +279,14 @@ private void search() {
                             }
                         }
                     }
+                    if (Type.Note.equals(reqType) && !pcc.allMatchesProcessed()) {
+                        String note = message.getHistoryRef().getHttpMessage().getNote();
+                        matcher = pattern.matcher(note);
+                        if (matcher.find()) {
+                            notifyMatchFound(currentRecordId, note, message, null, 0, 0);
+                            break;
```

Comment:
Does not need to break here (reason of https://github.com/zaproxy/zaproxy/pull/8536#issuecomment-2195211695).

Reply:
This was probably a copy/paste mistake from following the tag example. Since tags are looped we only need one match.

Reply:
This fixes the issue reported by @psiinon in https://github.com/zaproxy/zaproxy/pull/8536#issuecomment-2195211695.